src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -33,17 +33,15 @@ public class ModuleIntrinsics
         [TraceSource("Modules", "Module loading and analysis")]
```

Comment:
@iSazonov, your last commit had 3 failures in `PowerShell-CI-macos`
TabCompletion.Should complete abbreviated function
```powershell
Expected exactly 'PSConsoleHostReadLine', but got @('PSConsoleHostReadLine', 'PSReadLine\PSConsoleHostReadLine').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 22
22:         $res | Should -BeExactly 'PSConsoleHostReadLine'
```
Import-Module.should be able to add a module with using ModuleInfo switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
Import-Module.should be able to load a module with a trailing directory separator: @('/usr/local/microsoft/powershell/6/Modules/Microsoft.PowerShell.Security', '/Users/vsts/agent/2.155.1/work/1/a/bins/publish/Modules/Microsoft.PowerShell.Security', /)
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```